src/Microsoft.AspNet.Http/Features/FormFeature.cs

```diff
@@ -67,16 +68,6 @@ public bool HasFormContentType
 
         public IFormCollection ReadForm()
         {
-            if (Form != null)
```

Comment:
True. Will put back in the form take out the exception throw.

Why support the non-async method, when you clearly don't? Can avoid state machine with a slightly different pattern for cached results; but put the choice and onus of "Sync over Async" into user code?